File: Vehicle.java
// Abstract Parent Class
public abstract class Vehicle {

    protected int vehicleId;
    protected String vehicleNumber;
    protected int capacity;

    public Vehicle(int vehicleId, String vehicleNumber, int capacity) {

        this.vehicleId = vehicleId;
        this.vehicleNumber = vehicleNumber;
        this.capacity = capacity;
    }

    public abstract void displayVehicleDetails();
}

File: python/Vehicle.py
# Abstract Parent Class

from abc import ABC, abstractmethod


class Vehicle(ABC):
# allows us to create an Abstract Base Class.
    def __init__(self, vehicleId, vehicleNumber, capacity):
        self.vehicleId = vehicleId
        self.vehicleNumber = vehicleNumber
        self.capacity = capacity

    @abstractmethod
    def displayVehicleDetails(self):
        pass
